### PR TITLE
fix(BUX-250): fix decode;add spv and decode tests for corrupted/invalid beef

### PR DESCRIPTION
**Pull request by arkadiuszos4chain**
* fix BEEF decode - return errors on invalid BEEF
* add tests with invalid BEEFs
* add tests for SPV with corrupted BEEFs

 ** set version to `v0.7.2`** 

# Pull Request Checklist

- [x] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/go-paymail/blob/main/.github/CODE_STANDARDS.md)
- [x] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/go-paymail/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] 🏠 I tested my changes locally.
- [x] ✅ I have provided tests for my changes.
- [x] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [x] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: p2p_beef_tx.go [hunk 1]
```diff
@@ -91,17 +88,29 @@ func decodeBUMPs(beefBytes []byte) ([]BUMP, []byte, error) {
...
 	for i := 0; i < int(nBump); i++ {
 		if len(beefBytes) == 0 {
 			return nil, nil, errors.New("insufficient bytes to extract BUMP blockHeight")
 		}
 		blockHeight, bytesUsed := bt.NewVarIntFromBytes(beefBytes)
 		beefBytes = beefBytes[bytesUsed:]
-		bumpPaths, remainingBytes, err := decodeBUMPPathsFromStream(beefBytes)
+
+		treeHeight := beefBytes[0]
+		if int(treeHeight) > maxTreeHeight {
```
Comment (wregulski) [inline p2p_beef_tx.go:108]: Do we cover this path in unit tests? I think that we should. I mean when this error occurs.

File: p2p_beef_tx.go [hunk 2]
```diff
@@ -115,19 +124,17 @@ func decodeBUMPs(beefBytes []byte) ([]BUMP, []byte, error) {
 		bumps = append(bumps, bump)
 	}
 
+	if len(bumps) == 0 {
+		return nil, nil, errors.New("invalid BEEF- lack of BUMPs")
```
Comment (pawellewandowski98) [inline p2p_beef_tx.go:128]: Maybe another error message? This same is on line 94 so if the error appear  it will be unclear where exactly it came from

**Review by wregulski — CHANGES_REQUESTED** [review]

File: p2p_beef_tx.go (hunk 2 above)
Comment (wregulski) [reply p2p_beef_tx.go:128]: both should be handled in separated test cases in my opinion

Comment (arkadiuszos4chain) [reply p2p_beef_tx.go:128]: I realized this if statement has no sense - there is no option to got bumps collection shorter than declared Bumps without getting errors. I remove this

File: p2p_beef_tx.go (hunk 1 above)
Comment (arkadiuszos4chain) [reply p2p_beef_tx.go:108]: New test case in tests

**Review by pawellewandowski98 — APPROVED** [review]

**Review by wregulski — APPROVED** [review]

**Pull request merged**